snappy_test.go

```diff
@@ -322,6 +323,18 @@ func downloadTestdata(b *testing.B, basename string) (errRet error) {
 			os.Remove(filename)
 		}
 	}()
+	// Generate random data
+	if basename == "random" {
+		rng := rand.New(rand.NewSource(27354294))
+		for i := 0; i < 1<<20; i++ {
+			_, err := f.Write([]byte{byte(rng.Intn(255))})
```

Comment:
>One million syscalls and one million one-byte mallocs.

It is using `math/rand`, not `crypto/rand`, so there are no syscalls. It is a *one time* (the file is written to disk) operation. Optimizing this is *way* over the target.

I will update the limit to 256.